presto-main/src/main/java/com/facebook/presto/operator/InMemoryJoinHash.java

```diff
@@ -153,13 +174,22 @@ private int hashPosition(int position)
         return pagesHashStrategy.hashPosition(blockIndex, blockPosition);
     }
 
-    private boolean positionEqualsCurrentRow(int leftPosition, int rightPosition, Block... rightBlocks)
+    private boolean positionEqualsCurrentRow(int leftPosition, int rightPosition, Block[] rightHashBlocks, Block[] rightBlocks)
```

Comment:
why this method can not handle filter function? In other words, why can't you push down filter function to `pagesHashStrategy`?